### PR TITLE
test: fix and improve reboot/reset tests

### PR DESCRIPTION
**Pull request by smira**
These tests rely on node uptime checks. These checks are quite flaky.

Following fixes were applied:

* code was refactored as common method shared between reset/reboot tests
(reboot all nodes does checks in a different way, so it wasn't updated)

* each request to read uptime times out in 5 seconds, so that checks
don't wait forever when node is down (or connection is aborted)

* to account for node availability vs. lower uptime in the beginning of
test, add extra elapsed time to the check condition

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2245)

### CODE REVIEW

File: internal/integration/api/reboot.go [hunk 1]
```diff
@@ -143,20 +114,27 @@ func (suite *RebootSuite) TestRebootAllNodes() {
...
+					uptimeAfter, err := suite.ReadUptime(requestCtx)
 					if err != nil {
 						// API might be unresponsive during reboot
 						return retry.ExpectedError(fmt.Errorf("error reading uptime for node %q: %w", node, err))
 					}
 
-					if uptimeAfter >= uptimeBefore {
+					// uptime of the node before it actually reboots still goes up linearly
+					// so we can safely add elapsed time here
+					if uptimeAfter >= uptimeBefore+elapsed {
```
Comment (andrewrynhard) [inline internal/integration/api/reboot.go:135]: :+1:

**Comment by smira** [comment]
re-pushed with debugging `log.Printf` removed, no other changes

**Pull request merged**